h2quic/server.go

```diff
@@ -234,7 +248,30 @@ func (s *Server) handleRequest(session streamCreator, headerStream quic.Stream,
 			session.Close(nil)
 		}
 	}()
+	return nil
+}
 
+func (s *Server) handleSettingsFrame(h2SettingsFrame *http2.SettingsFrame, session streamCreator) error {
+	// PUSH
+	// TODO: in the working draft there is no EnablePush setting! This is replaced by a separate MAX_PUSH_ID frame
+	pushEnabled, ok := h2SettingsFrame.Value(http2.SettingEnablePush)
+	if ok {
+		s.pushEnabled[session.ConnectionID()] = (pushEnabled != 0)
+	}
+	// SETTINGS_HEADER_TABLE_SIZE
+	settingHeaderTableSize, ok := h2SettingsFrame.Value(http2.SettingHeaderTableSize)
+	if ok {
+		if settingHeaderTableSize != 0 {
+			// MUST be zero
+			return qerr.InternalError
```

Comment:
Is there a more specific error we can return here, maybe a `qerr.InvalidHeadersStreamData`?